client/reader/tag-stream/index.js

```diff
@@ -11,7 +12,14 @@ import { tagListing } from './controller';
 import { initAbTests, preloadReaderBundle, sidebar, updateLastRoute } from 'reader/controller';
 import { makeLayout, render as clientRender } from 'controller';
 
+const redirectHashtaggedTags = ( context, next ) => {
+	if ( context.hashstring && startsWith( context.pathname, '/tag/#' ) ) {
+		page.redirect( `/tag/${ context.hashstring }` );
+	}
+	next();
```

Comment:
just hitting `/tag` with no hash leaves you in a bad place. Could we redirect that to `/`?